### ISSUE TITLE
[Feature Request] Add Finder-like column view of folder

### ISSUE THREAD

**Issue by Sciancisco** [pop-os/cosmic-files#77]
Please add a column view à la Finder! I find it most efficient to visualize and navigate the filesystem.

Ever since switching to Linux, I missed that view. I also wonder why no DE except Pantheon (to my knowledge) implemented a column view.

**Comment by 12people** [comment]
I came here to report the exact same thing — Miller columns!

I switch between macOS and Linux and this is THE feature that makes using Nautilus so much more painful than Finder. (Unfortunately, this feature has been [rejected by the GNOME Nautilus team](https://gitlab.gnome.org/GNOME/nautilus/-/issues/27).)

Specifically, I really value being able to traverse complex folder hierarchies (useful esp. for development projects that have folders buried in folders buried in folders) with just my keyboard navigation keys without getting lost or disoriented.

Would love to see this in COSMIC Files.

**Comment by jackpot51** [comment]
We have strongly considered this, and it is very likely to be a feature.

**Comment by peastman** [comment]
I'm really glad to hear that!

If you're interested in considering some novel ideas, I think there's still room for innovation in file browsers.  I have an idea for how the column and list views could be combined into a single view with all the benefits of both.  Here's a mockup showing how I imagine it working.  This was created by pasting together pieces of a few different Finder views.

When you first open a file browser you would get a standard list view like in Finder or Nautilus.

![Screenshot 2024-04-09 at 5 29 11 PM](https://github.com/pop-os/cosmic-files/assets/4379786/a2532414-26e5-4130-a809-5275fe585af0)

You could do all the things you expect in a list view: select which fields to display, sort by any field, expand folders.  If you selected a file, an info panel would slide on from the right side, covering up everything except the Name column.

![image2](https://github.com/pop-os/cosmic-files/assets/4379786/20f968b7-cc77-4fd3-801d-d9500418d2a5)

If you deselected the file, the info panel would slide back off and you would get the full set of fields again.  If you instead selected a folder, a new list view would slide in from the right side, again covering up everything except the names.

![image3](https://github.com/pop-os/cosmic-files/assets/4379786/efe2d32c-f861-46af-a3eb-14c64f9213d5)

As you got deeper into the hierarchy, the columns for the outer directories would get pushed off the left side.  A horizontal scrollbar would let you scroll back through them.  So it's like the column view, but each column is itself hierarchical and sortable, and the innermost directory always shows the full set of fields from the list view.

Just something to think about!

**Comment by tveimo** [comment]
Miller columns would instantly be a great sales point for cosmic, and get you a lot of converts from other DEs.

**Comment by kmf** [comment]
I'm for this.

**Comment by lalishansh** [comment]
Hi guys,
I'd like to bring your attention to https://www.onecommander.com/ 
This one of the gorgeous miller column implementations (or file manager implementations)

(Looks way better in dark mode)
![image](https://github.com/user-attachments/assets/5b1a94f0-c884-47d2-9db5-f3b7d744c2a9)

**Comment by kmf** [comment]
@lalishansh have you seen https://github.com/euclio/fm written in Rust

**Comment by lalishansh** [comment]
> @lalishansh have you seen [euclio/fm](https://github.com/euclio/fm) written in Rust

yup, but never used it (i use gokcehan/lf for column like layout 😅)

**Comment by whos-bally** [comment]
Would genuinely be so happy if this feature was added, it's my most favourite & most missed feature of MacOS/OSX.

**Comment by Cloudwolf-dev** [comment]
Would love this. I've had many attempts last year trying to make Pantheon FIles work outside of Pantheon DE exclusively because of this feature, I only managed to do in NixOS because of a existing package on the nix repo but I stopped using Nix and went to PopOS and now I'm settled down on Cachy, only to miss this feature after trying to do work using my linux laptop.
